erica_snider/Assignments/Python/registration_and_login/server.py

```diff
@@ -0,0 +1,108 @@
+from flask import Flask, render_template, request, redirect, session, flash
+from mysqlconnection import MySQLConnector
+import re # the "re" module will let us perform some regular expression operations
+from flask.ext.bcrypt import Bcrypt
+app = Flask(__name__)
+mysql = MySQLConnector(app,'registration')
+app.secret_key = 'ThisIsSecret'
+EMAIL_REGEX = re.compile(r'^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z]+$')
+bcrypt = Bcrypt(app)
+
+def validate_first_name(first_name):
+    if len(first_name) < 1:
+        flash("First name cannot be empty.")
+        return False
+    else:
+        return True
+
+def validate_last_name(last_name):
+    if len(last_name) < 1:
+        flash("Last name cannot be empty.")
+        return False
+    else:
+        return True
+
+def validate_email(email):
+    if len(email) < 1:
+        flash("Email cannot be empty.")
+        return False
+    elif not EMAIL_REGEX.match(email):
+        flash("Email address must be valid.")
+        return False
+    else:
+        return True
+
+def validate_password(password):
+    if len(password) < 1:
+        flash("Password cannot be empty.")
+        return False
+    elif len(password) < 8:
+        flash("Password must be at least 8 characters.")
+        return False
+    else:
+        return True
+
+def validate_password_conf(password, password_conf):
+    if password != password_conf:
+        flash("Passwords must match.")
+        return False
+    return True
+
+@app.route('/')
+def index():
+    query = "SELECT * FROM users"
+    users = mysql.query_db(query)
+    return render_template('index.html', all_users=users)
+
+@app.route('/register', methods=['POST'])
+def create():
+    # Handle the registration form submit and create the user in the db
+    if validate_first_name(request.form['first_name']) and validate_last_name(request.form['last_name']) and validate_email(request.form['email']) and validate_password(request.form['password']) and validate_password_conf(request.form['password'], request.form['password_conf']):
```

Comment:
You can extract this out to a method, that checks all the values, and can make a cleaner if statement.